src/agent.py

```diff
@@ -53,6 +62,65 @@ def __init__(self, charm: ops.CharmBase, state: State):
         charm.framework.observe(
             charm.on[AGENT_RELATION].relation_departed, self._on_agent_relation_departed
         )
+        # Event hooks for agent-discovery-ingress
+        charm.framework.observe(
+            ingress_observer.ingress.on.ready,
+            self.reconfigure_agent_discovery,
+        )
+        charm.framework.observe(
+            ingress_observer.ingress.on.revoked,
+            self.reconfigure_agent_discovery,
+        )
+
+    @property
+    def agent_discovery_url(self) -> str:
+        """Return the external hostname to be passed to agents via the integration.
+
+        If we do not have an ingress, then use the pod ip as hostname.
+        The reason to prefer this over the pod name (which is the actual
+        hostname visible from the pod) or a K8s service, is that those
+        are routable virtually exclusively inside the cluster (as they rely)
```

Comment:
What is the reason for enclosing `as they rely` in brackets please?

Reply:
No particular reason as this was brought over from the prometheus charm and I didn't pay much attention to the docstring, I removed the brackets